apps/vax/lib/vax/adapter/antidote_client.ex

```diff
@@ -0,0 +1,41 @@
+defmodule Vax.Adapter.AntidoteClient do
+  @moduledoc false
+
+  @spec read_objects(
+          conn :: pid(),
+          objects :: [{key :: binary(), type :: atom(), bucket :: binary()}],
+          tx_id :: term()
+        ) :: {:ok, results :: [term]}
+  def read_objects(conn, objects, tx_id) do
+    metadata = %{objects: objects, tx_id: tx_id}
+
+    :telemetry.span([:vax, :read_objects], metadata, fn ->
+      {:antidotec_pb.read_objects(conn, objects, tx_id), metadata}
+    end)
+  end
+
+  @spec update_objects(conn :: pid(), ops :: list(), tx_id :: term()) :: :ok
+  def update_objects(conn, ops, tx_id) do
+    metadata = %{ops: ops, tx_id: tx_id}
+
+    :telemetry.span([:vax, :update_objects], metadata, fn ->
+      {:antidotec_pb.update_objects(conn, ops, tx_id), metadata}
+    end)
+  end
+
+  @spec start_transaction(conn :: pid(), last_tx_id :: binary() | :ignore, opts :: Keyword.t()) ::
+          {:ok, tx_id :: term()}
```

Comment:
Spec violates https://github.com/vaxine-io/antidote-erlang-client/blob/master/src/antidotec_pb.erl#L35 How do we handle error cases here?